fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/service/InterestPauseReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByExternalLoanIdAndTermType(
+                new ExternalId(loanExternalId), LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseResponse(variations);
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createInterestPause(ExternalId loanExternalId, LocalDate startDate, LocalDate endDate, String dateFormat,
+            String locale) {
+        this.context.authenticatedUser();
+
+        Long loanId = loanRepository.findIdByExternalId(loanExternalId);
+        if (loanId == null) {
+            throw new LoanNotFoundException("Loan not found with External ID: " + loanExternalId);
+        }
+
+        Loan loan = loanRepository.findByExternalId(loanExternalId).orElseThrow(() -> new LoanNotFoundException(loanExternalId));
+
+        validateInterestPauseDates(loan, startDate, endDate, dateFormat, locale);
+
+        LoanTermVariations variation = new LoanTermVariations(LoanTermVariationType.INTEREST_PAUSE.getValue(), startDate, null, endDate,
+                false, loan);
+
+        LoanTermVariations savedVariation = loanTermVariationsRepository.saveAndFlush(variation);
+
+        return new CommandProcessingResultBuilder().withEntityId(savedVariation.getId()).build();
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createInterestPauseByLoanId(Long loanId, LocalDate startDate, LocalDate endDate, String dateFormat,
+            String locale) {
+        this.context.authenticatedUser();
+
+        final Loan loan = loanRepository.findById(loanId).orElseThrow(() -> new LoanNotFoundException("Loan not found with ID: " + loanId));
+
+        validateInterestPauseDates(loan, startDate, endDate, dateFormat, locale);
+
+        LoanTermVariations variation = new LoanTermVariations(LoanTermVariationType.INTEREST_PAUSE.getValue(), startDate, null, endDate,
+                false, loan);
+
+        LoanTermVariations savedVariation = loanTermVariationsRepository.saveAndFlush(variation);
+
+        return new CommandProcessingResultBuilder().withEntityId(savedVariation.getId()).build();
+    }
+
+    private void validateInterestPauseDates(Loan loan, LocalDate startDate, LocalDate endDate, String dateFormat, String locale) {
+        final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+        final DataValidatorBuilder baseDataValidator = new DataValidatorBuilder(dataValidationErrors).resource("InterestPause");
+
+        baseDataValidator.reset().parameter("startDate").value(startDate).notBlank();
+        baseDataValidator.reset().parameter("endDate").value(endDate).notBlank();
+        baseDataValidator.reset().parameter("dateFormat").value(dateFormat).notBlank();
+        baseDataValidator.reset().parameter("locale").value(locale).notBlank();
```

Comment:
Missing the validations checker. This way it will not throw any exception...
Example:
```
validateOrThrow("loan", baseDataValidator -> {
      baseDataValidator.reset().parameter("startDate").value(startDate).notBlank();
}
```

Reply:
Fixed